server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java

```diff
@@ -2252,7 +2255,7 @@ private UserVm importKvmVirtualMachineFromDisk(final ImportSource importSource,
         UserVm userVm = null;
 
         Map<String, String> allDetails = new HashMap<>(details);
-        if (serviceOffering.isDynamic()) {
+        if (!serviceOffering.isDynamic()) {
             allDetails.put(VmDetailConstants.CPU_NUMBER, String.valueOf(serviceOffering.getCpu()));
```

Comment:
these are set for custom offering?

Reply:
Yes. service offering's cpu & memory is null for custom offerings. And value from details needs to be used to override the values for the service offering.

Reply:
are these details needed for dynamic offering, or non-dynamic offering ?

Reply:
This needs to be updated only for non-dynamic service offering. Because CPU & memory is not set in details for those cases.

 For a dynamic offering, this is already set in `details` from the user input.

Reply:
@vishesh92 
it is different from what I thought: if service offering is not dynamic, the CPU and Memory are fixed and no need to add record to user_vm_details for them.